lib/crypto/src/crypto.erl

```diff
@@ -2362,6 +2362,9 @@ format_pkey(rsa, Key) ->
     map_ensure_int_as_bin(Key);
 format_pkey(ecdsa, [Key, Curve]) ->
     {nif_curve_params(Curve), ensure_int_as_bin(Key)};
+format_pkey(eddsa, [PubKey, Curve]) when  Curve == ed25519;
+                                          Curve == ed448 ->
+    [ensure_int_as_bin(PubKey), Curve];
```

Comment:
Why is format_pkey returning a two-tuple for ecdsa and a list with 2 elements for eddsa, why not the same datatype always? And the order is different as well in the tuple Key is last and in the list Key is first?

Reply:
@HansN  can you comment to this, I just added what was missing to make it work.

Reply:
@KennethL asked:
> Why is format_pkey returning a two-tuple for ecdsa and a list with 2 elements for eddsa, why not the same datatype always? And the order is different as well in the tuple Key is last and in the list Key is first?

It is because the format of the Key down in the C-code is different for different algorithms.   That part of the crypto nif is a bit hairy, and would probably have been different if it had been designed from the very beginning to be extensible with by then unknown algorithms.  Now it is just a sort of most-easy-at-both-sides.

As soon as it is unpacked in C (pkey.c  static int get_pkey_private_key for example) it is quite well-structured and easy to follow. I also think that the amount of testing down in C is quite low.

What the added clause to format_pkey/2  (lines 2365-2367) adds, is that the PubKey now can be an large integer and not only an binary.  The reason is that in SSH the Key was already formatted as a list of a binary and an atom and returned untouched by the last clause of format_pkey/2, while the unpacking of stuff (certificates and such) in SSL may return a large integer.